src/cargo/ops/registry/publish.rs

```diff
@@ -160,7 +160,27 @@ pub fn publish(ws: &Workspace<'_>, opts: &PublishOpts<'_>) -> CargoResult<()> {
             .acquire_package_cache_lock(CacheLockMode::DownloadExclusive)?;
 
         for (pkg, _) in &pkgs {
-            verify_unpublished(pkg, &mut source, &source_ids, opts.dry_run, opts.gctx)?;
+            // Wrap verify_unpublished with enhanced error context
+            verify_unpublished(pkg, &mut source, &source_ids, opts.dry_run, opts.gctx)
+                .map_err(|e| {
+                    // Collect remaining packages that have not been published yet
+                    let remaining: Vec<_> = pkgs
+                        .iter()
+                        .filter(|(p, _)| p.package_id() != pkg.package_id())
+                        .map(|(p, _)| format!("{} v{}", p.name(), p.version()))
+                        .collect();
+                    let message = if !remaining.is_empty() {
+                        format!(
+                            "failed to publish `{}` v{}; the following crates have not been published yet: {}",
+                            pkg.name(),
+                            pkg.version(),
+                            remaining.join(", ")
+                        )
+                    } else {
+                        format!("failed to publish `{}` v{}", pkg.name(), pkg.version())
+                    };
+                    e.context(message)
+                })?;
```

Comment:
Do we need to extend the verify step with this information?

Part of the problem is that we can end up in a partially-published state when doing `transmit`.  At this point in the code, nothing is published